### PR TITLE
Fix all Rails 5.0 deprecations

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

**Comment by mmoll** [comment]
depends on GH-4867 (and includes it)

**Comment by xprazak2** [comment]
I tested, works. Nicely done!

File: app/models/role.rb [hunk 1]
```diff
@@ -289,8 +289,10 @@ def allowed_actions
   end
 
   def check_deletable
-    errors.add(:base, _("Cannot delete built-in role")) if builtin?
-    errors.empty?
+    if builtin?
+      errors.add(:base, _("Cannot delete built-in role")) if builtin?
```
Comment (ohadlevy) [inline app/models/role.rb]: no need for another if condition?

Comment (mmoll) [reply app/models/role.rb]: I'd say no, according to the tests and the naming of the method. ;)

Comment (ohadlevy) [reply app/models/role.rb]: but you have the same if a line above?

Comment (mmoll) [reply app/models/role.rb]: fixed.

**Comment by mmoll** [comment]
Katello failures seem due to https://github.com/Katello/katello/pull/7103 not yet beeing merged there, the rest is green, so this is ready to go, IMHO

**Comment by mmoll** [comment]
[test katello]

**Comment by mmoll** [comment]
all green now 💚

**Review by ohadlevy — APPROVED** [review]

**Comment by ohadlevy** [comment]
thank you @mmoll! and of course credit to @domcleal  as well.

**Pull request merged**